mrjob/conf.py

```diff
@@ -68,8 +67,8 @@ def default_options(self):
     def validated_options(self, opts, error_fmt):
         unrecognized_opts = set(opts) - self.ALLOWED_KEYS
         if unrecognized_opts:
-            log.warn(error_fmt % ', '.join(sorted(unrecognized_opts)))
-            return dict((k, v) for k, v in opts.iteritems()
+            log.warning(error_fmt % ', '.join(sorted(unrecognized_opts)))
```

Comment:
Okay, leaning towards fail-fast on bad log formatting strings (i.e., the way we do it now). We don't log enough stuff that there's any question of efficiency.